cmd/telegraf/telegraf.go

```diff
@@ -120,7 +120,7 @@ func runAgent(ctx context.Context,
 	c.OutputFilters = outputFilters
 	c.InputFilters = inputFilters
 	err := c.LoadConfig(*fConfig)
-	if err != nil {
+	if err != nil && *fConfigDirectory == "" {
```

Comment:
This would ignore parse errors in the initial config file, which should be fatal.  I think to fix this issue properly we need to split the functions reading the configuration files from the function parsing them.

Errors reading a file can be ignored/logged, so long as inputs+outputs are found.  But errors parsing a file should be reported and be fatal.

Reply:
Ooh, good point. Was #5573 more what you'd like to see?

Reply:
I think we need the split in responsibility in order to give the error messages we want, and the http config option complicates this further.  I can't really understand #5573, it might work but it doesn't have a logical flow to me.  Also, I would only use a bool in a function parameter if it were the only argument.

This is trickier than it looks, and I think we should put it on ice until we have time to revamp the config loading in depth.